tests/unit/test_check_operators/test_string_comparison.py

```diff
@@ -703,3 +703,153 @@ def test_not_matches_regex(data, comparator, dataset_type, expected_result):
         {"target": "target", "comparator": comparator}
     )
     assert result.equals(df.convert_to_series(expected_result))
+
+
+@pytest.mark.parametrize(
+    "data,separator,dataset_type,expected_result",
+    [
+        # SENDIG rule examples - valid interval date/times with equal precision
+        (
+            {
+                "target": [
+                    "2003-12-15T10:00/2003-12-15T10:30",
+                    "2003-12-01/2003-12-10",
+                    "2003-01-01/2003-06-30",
+                ]
+            },
```

Comment:
instead of defining a dictionary here like `{"target": [1,2,3]}`, you could just define the inner array like `[1,2,3]` and build the target below like:
`df = dataset_type.from_dict({"target": data})`
It just simplifies the test cases a bit.